sphinx/ext/autodoc/__init__.py

```diff
@@ -719,6 +727,21 @@ def is_filtered_inherited_member(name: str, obj: Any) -> bool:
                     doc = obj.docstring
 
                 doc, metadata = separate_metadata(doc)
+
+                # workaround for `TypedDict` subclassing; does not work with
+                # `typing_extensions.TypedDict` and requires explicitly passing
+                # `TypedDict` to subclasses
+                if not doc and is_typeddict(self.object):
+                    __annotations__ = self.get_attr(self.object, '__annotations__', {})
+                    if membername in __annotations__:
+                        original_bases = safe_getattr(self.object, '__orig_bases__', ())
+                        for base in original_bases:
+                            if (base.__qualname__, membername) in attr_docs:
+                                super_doc = attr_docs[(base.__qualname__, membername)]
+                                if super_doc:
+                                    attr_docs[(namespace, membername)] = super_doc
+                                    doc = '\n'.join(super_doc)
```

Comment:
To remove one level of indentation, you can directly use:

```py
if super_doc := attr_docs.get((base.__qualname__, membername), None):
	attr_docs.set_default((namespace, membername), super_doc)
``` 

Note that you should break after changing `doc` (hopefully, the original bases are in the correct order...). We could also check if `'\n'.join(super_doc)` renders as empty or not first before storing it.